apps/sms/test/marionette/thread_test.js

```diff
@@ -25,9 +32,74 @@ marionette('Thread Panel Tests', function() {
     );
   });
 
+  suite('General use cases', function() {
+    suite('Long SMS thread', function() {
+      var thread;
+      setup(function() {
+        thread = {
+          id: 1,
+          body: 'Thread message content',
+          lastMessageType: 'sms',
+          timestamp: Date.now(),
+          messages: [],
+          participants: ['TestMan']
+        };
+
+        for (var i = 1; i < 50; i++) {
+          thread.messages.push({
+            id: i,
+            iccId: null,
+            threadId: thread.id,
+            sender: null,
+            receiver: 'TestMan',
+            type: 'sms',
+            delivery: 'sent',
+            body: 'Thread message content ' + i,
+            timestamp: Date.now()
+          });
+        }
+
+        messagesApp.launch();
+
+        messagesApp.setStorage([thread]);
+      });
+
+      test('User can see all messages when scrolls up', function() {
+        var pageSize = 10;
+        var loadedPage = 1;
+
+        messagesApp.ThreadList.firstThread.tap();
+
+        // Composer panel should always be visible
+        assertIsDisplayed(messagesApp.Composer.sendButton);
+        assertIsDisplayed(messagesApp.Composer.messageInput);
+
+        thread.messages.forEach(function(message, index) {
+          var messageNode = messagesApp.Thread.findMessage(message.id);
+
+          // If current page is loaded, then message node should be visible
+          if (index < pageSize * loadedPage) {
```

Comment:
I'm afraid this is fragile; especially except the first page we don't load a fixed number of element... we just display whatever is already in the page.

So most of the time your logic will work, but we might have an intermittent issue.

I wonder if we could have an event when all the messages are added to the DOM ? `qa-messages-rendered` ?

In the (not so distant) future (because gerard-majax bugs me with this)  I'd like to not add everything to the DOM, that's why I think an event is better than a class... but it's maybe more difficult to handle from the test?